Client/src/Common/Submitter/BaseClientSubmitter.cs

```diff
@@ -355,6 +379,71 @@ public Output GetTaskOutputInfo(string taskId)
                            delayMs);
   }
 
+  private static async Task WaitForTasksAsync(Events.EventsClient eventsClient,
+                                              string              sessionId,
+                                              ICollection<string> taskIds,
+                                              ILogger             logger,
+                                              CancellationToken   cancellationToken)
+  {
+    var tasksNotFound = new HashSet<string>(taskIds);
+    using var streamingCall = eventsClient.GetEvents(new EventSubscriptionRequest
+                                                     {
+                                                       SessionId = sessionId,
+                                                       ReturnedEvents =
+                                                       {
+                                                         EventsEnum.TaskStatusUpdate,
+                                                         EventsEnum.NewTask,
+                                                       },
+                                                     });
+    try
+    {
+      while (await streamingCall.ResponseStream.MoveNext(CancellationToken.None))
+      {
+        cancellationToken.ThrowIfCancellationRequested();
```

Comment:
Why does the reading of the stream not use the cancellation token?
```suggestion
      while (await streamingCall.ResponseStream.MoveNext(cancellationToken))
      {
```